src/static/index.js

```diff
@@ -82,39 +82,43 @@ function clearReact() {
 $.getJSON('/user-config/config.json', (result) => {
   dashboardConfig = result;
 }).always(() => {
-  sessionManager = new SessionManager(dashboardConfig);
-
-  const parameters = {
-    config: dashboardConfig,
-    previousSession: JSON.stringify(sessionManager.getSession()),
-  };
-
-  // init app
-  app = elmApp.init({ flags: parameters });
-
-  const conf = sessionManager.getConfig();
-  astarteClient = new AstarteClient({
-    realmManagementUrl: conf.realmManagementApiUrl,
-    appengineUrl: conf.appEngineApiUrl,
-    pairingUrl: conf.pairingApiUrl,
-    flowUrl: conf.flowApiUrl,
-    enableFlowPreview: conf.enableFlowPreview,
-  });
-
-  if (sessionManager.isUserLoggedIn) {
-    astarteClient.setCredentials(sessionManager.getCredentials());
-  }
+  if (!dashboardConfig) {
```

Comment:
This little change creates a huge diff where everything is identical, just indented one layer deeper.
What about an early return?
```
if (!dashboardConfig) {
  app = elmApp.init();
  return;
}
```

Reply:
I just didn't want to duplicate the code needed to setup Elm ports, which follows the if-else statement.

Reply:
ports are just for page changes, but there should be no page changes in the editor only mode

Reply:
A quick experiment confirmed that those lines are also needed to render the react page, even if there is no subsequent route change.